### PR TITLE
Revert "[builtins] Try to ensure single copy of emulated TLS state"

### PR DESCRIPTION
**Pull request by edymtt**
This is a short term measure to let
`stdlib/symbol-visibility-darwin.test-sh` pass in Swift CI.

This reverts commit b8f04a670f27a84412099dd025fa762ee58f4c1a.

Addresses rdar://85587684

### CODE REVIEW

**Comment by edymtt** [comment]
@swift-ci test

**Review by compnerd — APPROVED** [review]
This can cause some issues for Android, though, they are likely to have encountered them already as this is a recent change.  I think that we should be avoiding building emulated TLS on Darwin instead.  That would be a fairly straightforward fix.

However, reverting to repair the CI temporarily makes sense.

**Comment by bnbarham** [comment]
If you want to test this before merging you'd have to open a PR in Swift on top of rebranch and then reference that in the comment. I'm happy to merge this if you've verified it fixes it though.

EDIT: Turns out CI actually handles this case so... ignore me. Looks like there are actually compilation failures on Linux.

**Comment by bnbarham** [comment]
https://github.com/apple/swift/pull/40345
@swift-ci please test

**Comment by bnbarham** [comment]
Only failure on macos was `Swift(macosx-x86_64) :: Sanitizers/tsan/racy_async_let_fibonacci.swift` which is already known (and notably `Swift(macosx-x86_64) :: stdlib/symbol-visibility-darwin.test-sh` did *not* fail, which is the one we care about here). I'd say good to merge!

**Review by bnbarham — APPROVED** [review]

**Pull request merged**
